zoom_rooms/indico_zoom_rooms/plugin.py

```diff
@@ -0,0 +1,271 @@
+
+
+def _signal_link_object_updated(obj: Session | SessionBlock | Contribution | Event, changes: dict | None = None):
+    _check_link_object_for_updates(obj, changes)
+
+
+def _signal_tt_entry_updated(obj_type, entry: TimetableEntry, obj: Contribution | SessionBlock, changes: dict):
+    _check_link_object_for_updates(obj, changes)
+
+
+def _signal_zoom_meeting_created(vc_room: VCRoom, assoc: VCRoomEventAssociation, event: Event):
+    _handle_link_object_created(assoc)
+
+
+def _signal_zoom_meeting_cloned(
+    old_assoc: VCRoomEventAssociation,
+    new_assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    link_object: SessionBlock | Contribution | Event,
+):
+    # on clone, clone also the zoom room calendar entry
+    if new_assoc.link_object.room:
+        _handle_link_object_created(new_assoc)
+
+
+def _signal_zoom_meeting_association_attached(
+    assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    event: Event,
+    data: dict,
+    old_link: VCRoomEventAssociation | None,
+    new_room: bool = False,
+):
+    # ignore associations to new rooms, since that's already handled by `new_vc_room`
+    if not new_room:
+        # this is a new association to an existing room, hence a new slot has to be created
+        _handle_link_object_created(assoc)
+
+
+def _signal_zoom_meeting_association_detached(
+    assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    old_link: Event | Contribution | SessionBlock,
+    event: Event,
+    data: dict | None = None,
+):
+    if vc_room.type == 'zoom':
+        # when a room is detached from an event/contribution/session block, delete the corresponding zoom room entry
+        _delete_zoom_association(old_link, vc_room)
+
+
+def _signal_zoom_meeting_data_updated(vc_room: VCRoom, data: dict):
+    if vc_room.type == 'zoom' and (name := data.get('name')) and name != vc_room.name:
+        for assoc in vc_room.events:
+            if zr_id := get_zoom_room_id(assoc.link_object):
+                ZoomRoomsQueueEntry.record(ZoomRoomsAction.update, zr_id, assoc=assoc, args=OperationArgs(title=name))
+
+
+class ZoomRoomsPlugin(IndicoPlugin):
+    """Zoom Rooms
+
+    Zoom Rooms / Exchange synchronization plugin for Indico."""
+
+    configurable = True
+    settings_form = PluginSettingsForm
+    default_settings = {'debug': False, 'service_url': None, 'docs_url': '', 'token': None, 'timeout': 3}
+    category = PluginCategory.videoconference
+
+    def init(self):
+        super().init()
+        # Here we plug the various object signals into the action tracking code
+        # Consult the README.md for a summary of the logic behind
+        self.connect(signals.event.contribution_updated, _signal_link_object_updated)
+        self.connect(signals.event.session_block_updated, _signal_link_object_updated)
+        self.connect(signals.event.updated, _signal_link_object_updated)
+        self.connect(signals.event.times_changed, _signal_tt_entry_updated, sender=Contribution)
```

Comment:
Does it need to be handled for `sender=Event` as well?

Reply:
No, because that's already caught by `signals.event.updated` (line above).

Reply:
I just tested and it's indeed the case. So it needs to be handled for `Event` as well, and possibly the `_signal_link_object_updated` handler should ignore time-related updates to avoid recording them twice in cases where both signals are triggered.

Reply:
Yes, you are right. I hadn't thought of that case. Ufff.

Reply:
I think `times_changed` is only called in `track_time_changes`, so there's no need to do deduplication.

Reply:
What I mean is that when you edit the event time via the general event settings, both signals get triggered. But removing times from the event updated signals also has the advantage that people are forced to use the correct signal.